.circleci/config.yml

```diff
@@ -84,23 +85,23 @@ jobs:
             # If nixpkgs changes then potentially a lot of cached packages for
             # the base system will be invalidated so we may as well drop them
             # and make a new cache with the new packages.
-            - paymentserver-nix-store-v3-{{ checksum "nixpkgs.rev" }}-{{ checksum "nix/challenge-bypass-ristretto-repo.nix" }}
-            - paymentserver-nix-store-v3-{{ checksum "nixpkgs.rev" }}-
-            - paymentserver-nix-store-v3-
+            - paymentserver-nix-store-v5-{{ checksum "nixpkgs.rev" }}-{{ checksum "nix/challenge-bypass-ristretto-repo.nix" }}
+            - paymentserver-nix-store-v5-{{ checksum "nixpkgs.rev" }}-
+            - paymentserver-nix-store-v5-
 
       - run:
           name: "Building with Nix"
           command: |
             nix-build \
-              --option extra-substituters https://hydra.iohk.io/ \
+              -j 4 \
               ./nix/ \
               -A PaymentServer.components.exes."PaymentServer-exe"
 
       - run:
           name: "Building Tests"
           command: |
             nix-build \
-              --option extra-substituters https://hydra.iohk.io/ \
+              -j 4 \
```

Comment:
xlarge has 8 vCPUs.  If the cache hits then I doubt -j4 vs -j8 is going to matter much.  If the cache misses I wonder if even xlarge has enough RAM (16GB) to stand -j8.  I could wonder the same about -j4 I guess... Hm.  So much fun trying to juggle resource constraints in multiple dimensions with a single parameter and with "your build was killed" as the only failure mode.